Sources/ComposableArchitecture/Store.swift

```diff
@@ -1092,7 +1096,11 @@ extension ScopedStoreReducer: AnyScopedStoreReducer {
         Logger.shared.log("\(storeTypeName(of: store)).scope")
       }
     if let id = id {
-      store.children[id] = childStore
+      if store.canCacheChildren {
+        store.children[id] = childStore
+      } else {
+        // runtimeWarn("Can't cache child store at '\(id)' in an uncached store")
```

Comment:
This runtime warning isn't super helpful while the library still has `WithViewStore`s that scope in an uncache-able way.

```suggestion
      } else {
        // runtimeWarn("Can't cache child store at '\(id)' in an uncached store")
```